### PR TITLE
Regression sensitivity rewrite

### PR DESCRIPTION
**Pull request by platawiec**
This PR does the following:

- #185: Revamp `regression_sensitivity`. Now the user passes `Regression()` to `gsa`.
  - Seems to work and tests pass, works well with multi-input and multi-output
- #99: Drops dependencies on Dataframes and GLM
- #15: Samples for regression analysis generated using Sobol sampling from QuasiMonteCarlo
- #166: Uses a kwarg `samples

Would appreciate a second pair of eyes and any requests or suggestions on naming or the user-facing API!

### CODE REVIEW

File: src/DiffEqSensitivity.jl [hunk 1]
```diff
@@ -30,7 +30,7 @@ export extract_local_sensitivities
 export ODEForwardSensitivityFunction, ODEForwardSensitivityProblem, SensitivityFunction,
        ODEAdjointSensitivityProblem, ODEAdjointProblem, AdjointSensitivityIntegrand,
        adjoint_sensitivities, adjoint_sensitivities_u0, Sobol, Morris, gsa,
-       SensitivityAlg, regression_sensitivity, DGSM, eFAST
+       SensitivityAlg, Regression, DGSM, eFAST
```
Comment (ChrisRackauckas) [inline src/DiffEqSensitivity.jl]: That's probably too common of a name to export.

**Comment by ChrisRackauckas** [comment]
It looks good to be, but lets see if @Vaibhavdixit02 has any eyeballs rolling around his apartment.

File: src/DiffEqSensitivity.jl (hunk 1 above)
Comment (Vaibhavdixit02) [reply src/DiffEqSensitivity.jl]:
```suggestion
       SensitivityAlg, RegressionGSA, DGSM, eFAST
```
?

**Comment by Vaibhavdixit02** [comment]
Looks good to me, will merge once the name change is taken care of, `Regression` really is very generic.

**Comment by ChrisRackauckas** [comment]
Can we get a follow-up doc PR?

**Comment by platawiec** [comment]
Yes, planning on it, eta 1 week.

**Comment by ChrisRackauckas** [comment]
Thanks!

**Pull request merged**